### PR TITLE
Feature/add competition escrow

### PR DESCRIPTION
**Pull request by achillegley**
Change Competition winner selection and payment process to escrow contract. 

Admin create prompt and deposit fund for the prompt.
Admin select a winner ===> setRecipient for the escrowContract.
The selected winner claimed the matic paid..

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -41,3 +41,7 @@ coverage/
 conf-example
 functions/node_modules
 package-lock.json
+
+package-lock.json
+package.json
+quasar.config.js
```
Comment (hermannleboss) [inline .gitignore:47]: I don't get why you ignore package.json and quasar.config
Can you explain ?

File: src/components/Admin/FundDepositCard.vue [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+  }
+}
+
+async function onSubmit(event) {
+  event.preventDefault()
+
+  $q.loading.show()
+  try {
+    const result = await depositFunds({ amountInMatic: maticAmount.value })
+    // const result = {
```
Comment (marclupanc) [inline src/components/Admin/FundDepositCard.vue:78]: Please remove all the commented code

File: src/components/Admin/ManagePromptsEntries.vue [hunk 3]
```diff
@@ -249,4 +283,21 @@ function handleDeleteEntry(entryId, promptId) {
 
   promptStore.fetchPrompts()
 }
+
+//proceed deposit funds.
+async function onProceedDepositFundDialog(props) {
+  //let's check if the entry already have valid payment..
+  //console.log('the current wallet address ', currentWalletAddress)
```
Comment (marclupanc) [inline src/components/Admin/ManagePromptsEntries.vue:290]: And console logs

File: src/components/Admin/ManagePromptsEntries.vue [hunk 4]
```diff
@@ -249,4 +283,21 @@ function handleDeleteEntry(entryId, promptId) {
...
+    customWeb3modal.open()
+  } else {
+    proceedDepositFundDialog.value.show = true
+    proceedDepositFundDialog.value.walletAddress = customWeb3modal.getAddress()
+    proceedDepositFundDialog.value.prompt = props
+  }
+}
+
+//deposit funds function
+//const _depositFunds= async(payload={amountInMatic})
```
Comment (marclupanc) [inline src/components/Admin/ManagePromptsEntries.vue:302]: To remove

File: src/components/Admin/TableEntry.vue [hunk 5]
```diff
@@ -237,24 +231,48 @@ function onDeleteDialog(entry) {
...
+  if (!_currentPrompt.value && props?.prompt?.id) {
+    _currentPrompt.value = await promptStore.fetchPromptById(props.prompt.id)
+  }
+  if (_currentPrompt.value) {
+    $q.loading.show()
+    if (!customWeb3modal.getAddress()) {
+      $q.notify({ type: 'negative', message: ' please connect your wallet ' })
+      customWeb3modal.open()
+      $q.loading.hide()
+      // onProceedPaymentDialog.value.show = false
```
Comment (marclupanc) [inline src/components/Admin/TableEntry.vue:243]: To remove

File: src/components/Admin/TableEntry.vue [hunk 6]
```diff
@@ -237,24 +231,48 @@ function onDeleteDialog(entry) {
...
+        displayCrytptoTransactionDialog.value.cryptoTransaction = cryptoTransactionExist[0]
+        displayCrytptoTransactionDialog.value.show = true
+      } else {
+        if (!props.author.walletAddress) {
+          $q.notify({ type: 'negative', message: ' the entry author should set wallet address ' })
+        } else {
+          const escrowEvents = await getEventsForEscrow({ escrowId: _currentPrompt.value.escrowId })
+
+          if (escrowEvents?.status?.includes('success')) {
+            //proceedPaymentDialog.value.depositedAmount = 0
```
Comment (marclupanc) [inline src/components/Admin/TableEntry.vue:257]: To remove

File: src/components/Admin/WalletPaymentCard.vue [hunk 7]
```diff
@@ -3,20 +3,20 @@
     <q-card-section class="q-pt-none">
       <q-form @submit="onSubmit">
         <q-input hide-hint label="Winner wallet address" maxlength="80" required v-model="_walletAddress" disable />
-        <q-input
+        <!-- <q-input
```
Comment (marclupanc) [inline src/components/Admin/WalletPaymentCard.vue:6]: To remove

File: src/components/Advertiser/AdvertiserTable.vue [hunk 8]
```diff
@@ -383,7 +383,7 @@ async function _claimPayment(advertise, currentAmountSpent) {
   $q.loading.show()
   const result = await claimPayment({ campaignCode: advertise.campaignCode, currentAmounSpentInMatic: currentAmountSpent })
   if (result.status.includes('success')) {
-    console.log('the result claimPayment result ====', result)
+    //console.log('the result claimPayment result ====', result)
```
Comment (marclupanc) [inline src/components/Advertiser/AdvertiserTable.vue:386]: To remove

**Pull request closed without merging**